libraries/AP_HAL_ChibiOS/hwdef/common/board.c

```diff
@@ -233,6 +234,10 @@ void __early_init(void) {
 void __late_init(void) {
   halInit();
   chSysInit();
+  stm32_watchdog_save_reason();
+#ifndef HAL_BOOTLOADER_BUILD
+  stm32_watchdog_clear_reason();
```

Comment:
Wouldn't we want to do this normally? IE if we had an old bootloader that wasn't updated?

Reply:
bootloader must not clear the reason, or main fw can't see it